src/util/PromiseUtil.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * A replacement for the Promise.any() function, which is only supported
+ * by NodeJs starting from version 15.0.0.
+ *
+ * @remarks
+ *
+ * Predicates provided as input must be implemented considering
+ * the following considerations:
+ * 1. if they throw an error, it won't be propagated;
+ * 2. throwing an error should be logically equivalent to return false.
```

Comment:
This function is not an identical replacement for `Promise.any`. Which is fine (we need different functionality), but the text is confusing.

Given the we are implementing more like the equivalent of `Array.some` for promises, I wonder if we can

```javascript
export function promiseSome(predicates) {
  return new Promise((resolve, reject) => {
    for (const predicate of predicates) {
      predicate.then(value => value && resolve(true), reject);
    }
    Promise.all(predicates).then(values => resolve(false), reject);
  });
}
```

to also no lose errors.

Reply:
If promiseSome propagates errors, then it will appear another little problem: there are legitimate situations in which accessChecker.handleSafe() can throw a NotImplementedHttpError (i.e. when the given acl rule does not contain a triple with predicate acl:agentGroup). This particular case should be handled.

Anyway, I see that an overall refactoring of the WebAclAuthorizer could be made to avoid certain functions to be executed potentially many times. Just an example: `createAuthorization` calls 2 times `determinePermissions`, which call 4 times `hasPermission`. `hasPermission` calls `hasAccess` for every rule that refers to a particular access mode (these rules could potentially be A LOT). Then, `hasAccess` calls once the accessChecker, which in turn calls `isMemberOfGroup` for every group referenced by a particular rule (these groups could, again, be potentially A LOT).

Considering the worst case, some lines of code could be executed 2x4xNxM times. It feels like some of this complexity can be reduced, hopefully leading to code easier to follow and understand. I'll take a look and reason about it in the following days.

Reply:
> there are legitimate situations in which accessChecker.handleSafe() can throw a NotImplementedHttpError

Right, then we probably want to `.catch(x => false)` them